lighthouse-core/gather/gather-runner.js

```diff
@@ -197,7 +183,7 @@ class GatherRunner {
       passContext.options = gathererDefn.options || {};
       const artifactPromise = Promise.resolve().then(_ => gatherer.beforePass(passContext));
       gathererResults[gatherer.name] = [artifactPromise];
-      await GatherRunner.recoverOrThrow(artifactPromise);
+      await artifactPromise;
```

Comment:
Sorry, I looked too quickly :) I should have said it'll have to be something like `await artifactPromise.catch(() => {});` (basically the equivalent of deleting just the `if (err.fatal) {throw err;}` part of `recoverOrThrow`).

`gathererResults[gatherer.name]` needs to be set to the raw promise, which may resolve or reject, which will become either the artifact or an artifact error in `collectArtifacts()`. But the `await` needs to never throw so the failure of one gatherer won't prevent moving on to the next